forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4505,14 +4506,38 @@ public final void setTapped(boolean tapped0) {
         view.updateTapped(this);
     }
 
+    public final boolean canTap() {
+        return canTap(false);
+    }
+    public final boolean canTap(boolean attacker) {
+        if (tapped) { return false; }
+
+        // Check replacement effects
+        Map<AbilityKey, Object> repParams = AbilityKey.mapFromAffected(this);
+        repParams.put(AbilityKey.IsCombat, attacker); // right name for parameter?
+        List<ReplacementEffect> list = getGame().getReplacementHandler().getReplacementList(ReplacementType.Tap, repParams, ReplacementLayer.CantHappen);
+
+        return list.isEmpty();
+    }
+
     public final boolean tap(boolean tapAnimation, SpellAbility cause, Player tapper) {
         return tap(false, tapAnimation, cause, tapper);
     }
     public final boolean tap(boolean attacker, boolean tapAnimation, SpellAbility cause, Player tapper) {
         if (tapped) { return false; }
 
         // Run replacement effects
-        getGame().getReplacementHandler().run(ReplacementType.Tap, AbilityKey.mapFromAffected(this));
+        Map<AbilityKey, Object> repParams = AbilityKey.mapFromAffected(this);
+        repParams.put(AbilityKey.IsCombat, attacker); // right name for parameter?
+
+        switch (getGame().getReplacementHandler().run(ReplacementType.Tap, repParams)) {
+        case NotReplaced:
+            break;
+        case Updated:
```

Comment:
it was previously used by _Illusionary Mask_ so you probably need to that change that into Updated result to work correctly

Reply:
That's why Updated is handled the same way as not replaced

Reply:
you need to also update it here because it is not the default result:
https://github.com/Card-Forge/forge/blob/be2f9aa187a3b6d06f1608a66085c3173d9b0560/forge-game/src/main/java/forge/game/ability/effects/PlayEffect.java#L564